File: app/src/main/java/com/example/plantea/dominio/Planificacion.java
package com.example.plantea.dominio;

import android.app.Activity;

import java.io.Serializable;
import java.util.ArrayList;

public class Planificacion implements Serializable {

    private String titulo;
    private int id;
    private boolean resultado;
    ArrayList<Planificacion> listaPlanes;
    ArrayList<Pictograma> listaPictogramas;
    GestionPlanificaciones gestionPlan = new GestionPlanificaciones();

    public Planificacion(){
    }
    public Planificacion(String titulo, int id) {
        this.titulo = titulo;
        this.id = id;
    }

    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public boolean crearPlanificacion(Activity actividad, ArrayList<Pictograma> pictogramas, String titulo) {
        resultado = gestionPlan.insertarPictogramaPlan(actividad,pictogramas, titulo);
        return resultado;
    }

    public ArrayList mostrarPlanificacionesDisponibles(Activity actividad) {
        listaPlanes = new ArrayList<>();
        listaPlanes = gestionPlan.listarPlanificaciones(actividad);
        return listaPlanes;
    }

    public void eliminarPlanificacion(Activity actividad, int id_plan) {
        gestionPlan.eliminarPlanificacion(actividad, id_plan);
    }

    public ArrayList obtenerPictogramasPlanificacion(Activity actividad, int id_plan) {
        listaPictogramas = new ArrayList<>();
        listaPictogramas = gestionPlan.obtenerPictogramasPlanificacion(actividad, id_plan);
        return listaPictogramas;
    }
    public void actualizarPlanificacion(Activity actividad, int id_plan, String nombre, ArrayList<Pictograma> pictogramas) {
        gestionPlan.actualizarPlanificacion(actividad, id_plan, nombre, pictogramas);
    }

    //Mostrar la planificacion a seguir
    public ArrayList mostrarPlanificacion(Activity actividad) {
        listaPlanes = new ArrayList<>();
        listaPlanes = gestionPlan.obtenerPictogramas(actividad);
        return listaPlanes;
    }

    //Obtener el titulo de la planificacion a seguir
    public String obtenerTituloPlan(Activity actividad) {
        titulo = gestionPlan.obtenerTituloPlan(actividad);
        return titulo;
    }
}

File: app/src/main/java/com/example/plantea/dominio/Planificacion.kt
package com.example.plantea.dominio

import android.app.Activity
import android.content.Context
import java.io.Serializable

class Planificacion : Serializable {
    private lateinit var titulo: String
    private var id = 0
    lateinit var listaPictogramas: ArrayList<Pictograma>
    private var gestionPlan = GestionPlanificaciones()

    data class PlanificacionItem(val title: String, var date: String)

    constructor()

    constructor(titulo: String, id: Int) {
        this.titulo = titulo
        this.id = id
    }

    // Getters
    fun getTitulo() = titulo
    fun getId() = id

    fun setTitulo(titulo: String) {
        this.titulo = titulo
    }

    fun crearPlanificacion(actividad: Activity?,idUsuario: String, titulo: String?): Int {
        return gestionPlan.insertarPlanificacion(actividad, idUsuario, titulo)
    }

    fun addPictogramasPlan(idPlan: Int?,actividad: Activity?, idUsuario: String?, listaPlanificacion: ArrayList<Pictograma>): Any {
        return gestionPlan.addPictogramasPlan(actividad, idPlan, idUsuario, listaPlanificacion)
    }

    fun addPictogramasPlanTraductor(idPlan: Int?, actividad: Activity?, listaPlanificacion: ArrayList<Pictograma>, idUsuario: String?): Any {
        return gestionPlan.addPictogramasPlanTraductor(actividad, idPlan, listaPlanificacion, idUsuario)
    }

    fun mostrarPlanificacionesDisponibles(idUsuario: String, actividad: Activity?): ArrayList<*> {
       // return gestionPlan.listarPlanificaciones(idUsuario, actividad) as ArrayList<Planificacion>
        return gestionPlan.listarPlanificaciones(idUsuario, actividad)
    }

    fun eliminarPlanificacion(actividad: Activity?, idPlan: Int) {
        gestionPlan.eliminarPlanificacion(actividad, idPlan)
    }

    fun obtenerPictogramasPlanificacion(actividad: Activity?, idPlan: Int, language: String, idUsuario: String): ArrayList<Pictograma> {
        return gestionPlan.obtenerPictogramasPlanificacion(actividad, idPlan, language, idUsuario)
    }

    fun actualizarPlanificacion(actividad: Activity?, idUsuario: String, idPlan: Int, nombre: String?, pictogramas: ArrayList<Pictograma>) {
        gestionPlan.actualizarPlanificacion(actividad, idUsuario, idPlan, nombre, pictogramas)
    }

    //Mostrar la planificacion a seguir
    fun mostrarPlanificacion(idUsuario: String, id: String, context: Context?, language: String): ArrayList<*> {
        return gestionPlan.obtenerPictogramas(idUsuario, id, context, language) as ArrayList<Planificacion>
    }

    fun obtenerTitulosPlanificaciones(idUsuario: String, actividad: Activity?): ArrayList<String> {
        return gestionPlan.obtenerTitulosPlanificaciones(idUsuario, actividad)
    }



}